addons/stock/models/stock_location.py

```diff
@@ -177,6 +178,29 @@ def _get_putaway_strategy(self, product):
             current_location = current_location.location_id
         return putaway_location
 
+    def _get_next_inventory_date(self):
+        """ Used to get the next inventory date for a quant located in this location. It is
+        based on:
+        1. Does the location have a cyclic inventory set?
+        2. If not 1, then is there an annual inventory date set (for its company)?
+        3. If not 1 and 2, then quants have no next inventory date."""
+        next_inventory_date = False
+        if self.next_inventory_date:
+            next_inventory_date = self.next_inventory_date
+        elif self.company_id.annual_inventory_month:
+            annual_inventory_month = int(self.company_id.annual_inventory_month)
+            annual_inventory_day = self.company_id.annual_inventory_day
+            if annual_inventory_day <= 0:
+                annual_inventory_day = 1
+            # check if larger than max day for month in a non-leap year
+            max_day = calendar.monthrange(2021, annual_inventory_month)[1]
```

Comment:
Why harcoded 2021 ? If someone wants use the 29 February of 2024 :cry:

Reply:
Because wouldn't it be weird to have your annual inventory on Feb 28th except every 4th year?

Reply:
Mhh good point

Reply:
Expect if you just want the last day of Feb each year

Reply:
Also a good point, but I'll have to add extra hardcoded logic to handle when someone does an inventory on Feb 29 so that the `next_inventory_date = today.replace(year=next_inventory_date.year + 1)` assigns to Feb 28th instead... Is it worth it? Something like:
```
if next_inventory_date == today:
    if not (today.month == 2 and today.day == 29):
        next_inventory_date = today.replace(year=next_inventory_date.year + 1)
    else:
        next_inventory_date = today.replace(day=28, year=next_inventory_date.year + 1)
```

Reply:
More like: 
```python
...
today = fields.Date.today()   # Call only one time
year = today.year + 1 if today.month == annual_inventory_month and today.day == annual_inventory_day else today.year
max_day = calendar.monthrange(year, annual_inventory_month)[1]
...
```

Reply:
Ughh, just kidding I realized a couple other mistakes + handling the leap year is even more annoying than I thought. Logic should be good now.